packages/expo-system-ui/src/SystemUI.ts

```diff
@@ -1,7 +1,20 @@
+import Constants from 'expo-constants';
 import { ColorValue, Platform, processColor } from 'react-native';
 
 import ExpoSystemUI from './ExpoSystemUI';
 
+declare global {
+  /**
+   * This variable is set to true when edge to edge mode is enabled
+   * @example
+   * if (__EDGE_TO_EDGE__) console.log('Edge to edge mode is enabled')
+   */
+  const __EDGE_TO_EDGE__: boolean;
+}
+
+// @ts-ignore
+global.__EDGE_TO_EDGE__ = Constants.expoConfig?.experiments?.edgeToEdge ?? false;
```

Comment:
Its purpose is only to be used by libraries to determine if edge-to-edge is enabled and provides better defaults. For example, this could be used in `react-native-reanimated` like this:

```ts
declare global {
  const __EDGE_TO_EDGE__: boolean | undefined;
}

const useAnimatedKeyboard = (options?: {
  isStatusBarTranslucentAndroid?: boolean;
}) => {
  const isStatusBarTranslucentAndroid =
    options?.isStatusBarTranslucentAndroid ?? __EDGE_TO_EDGE__ ?? false;

  // …
};
```

I can't think of another way to properly provides this feature in Expo 😕

I guess we can avoid adding it and wait for edge-to-edge being implemented directly in `react-native`, where its place could makes more sense (next to `__DEV__` declaration).

Reply:
what if we made a library called `react-native-edge-to-edge`, which you could use like:  `import * as EdgeToEdge from 'react-native-edge-to-edge'` -> `EdgeToEdge.isEnabled(): boolean`

then we can call this whatever we want, such as `__EDGE_TO_EDGE_EXPERIMENTAL_DO_NOT_USE_THIS_VARIABLE_DIRECTLY__`. then any library that wants to add edge to edge support includes a dependency on this, so we can easily change up how it's detected at any time

Reply:
@brentvatne Having the experiment in a lib is the best of both worlds, but in this case it's probably better to put all the logic in it, and only document it in the expo documentation.

And as a huge bonus, this will be easily usable by everyone, not only Expo users.